src/user/user.controller.ts

```diff
@@ -0,0 +1,14 @@
+import {
+  Controller,
+  Get,
+  Param,
+  Post,
+  Body,
+  Put,
+  Delete,
+} from '@nestjs/common';
+import { UserService } from './user.service';
+import { User as UserModel } from '@prisma/client';
```

Comment:
Hmm, I think controller should not have imports to prisma - see if you will be able to remove this import after you have implemented the approach I've mentioned above about response DTOs classes